contrib/mesos/pkg/election/master.go

```diff
@@ -88,10 +88,12 @@ func Notify(m MasterElector, path, id string, s Service, abort <-chan struct{})
 						}
 						func() {
 							n.lock.Lock()
-							defer n.lock.Unlock()
 							n.desired = electedMaster
-							if n.desired != n.current {
-								n.cond.Signal()
+							newMaster := n.desired != n.current
+							n.lock.Unlock()
+
+							if newMaster {
+								n.changed <- struct{}{}
```

Comment:
Wondering whether this blocking call is really indented. I think we only have to make sure that the last "changed" event is not dropped.